lib/inv_cg_quda.cpp

```diff
@@ -324,16 +325,33 @@ namespace quda {
     }
     blas::zero(x);
     if (&x != &xSloppy) blas::zero(xSloppy);
-
     blas::copy(rSloppy,r);
+
     if (Np != (int)p.size()) {
       for (auto &pi : p) delete pi;
       p.resize(Np);
       ColorSpinorParam csParam(rSloppy);
       csParam.create = QUDA_COPY_FIELD_CREATE;
-      for (auto &pi : p) pi = ColorSpinorField::Create(rSloppy, csParam);
+      if(pin){
+        for (auto &pi : p) pi = ColorSpinorField::Create(*pin, csParam);        
+      } else {
+        for (auto &pi : p) pi = ColorSpinorField::Create(rSloppy, csParam);
+      }
     } else {
-      for (auto &pi : p) *pi = rSloppy;
+      if(pin){
```

Comment:
Why not just write this as?
```c++
for (auto &p_i : p) *p_i = p_in ? *p_in : rSloppy;
```
Or at least why not just use `=` instead of `blas::copy` for the first case?

Also, style police: for consistency  `if( ){` -> `if () {`